sdk/go/x/auto/example_test.go

```diff
@@ -1073,3 +1075,98 @@ func ExampleStack_SetAllConfig() {
 	// set all config in map
 	_ = stack.SetAllConfig(ctx, cfg)
 }
+
+func ExampleStack_Cancel() {
+	ctx := context.Background()
+	stackName := FullyQualifiedStackName("org", "project", "stack")
+	stack, _ := SelectStackLocalSource(ctx, stackName, filepath.Join(".", "program"))
+	// attempt to cancel the in progress operation
+	// note that this operation is _very dangerous_, and may leave the stack in an inconsistent state.
+	_ = stack.Cancel(ctx)
+}
+
+func ExampleStack_Export() {
+	ctx := context.Background()
+	stackName := FullyQualifiedStackName("org", "project", "stack")
+	stack, _ := SelectStackLocalSource(ctx, stackName, filepath.Join(".", "program"))
+	dep, _ := stack.Export(ctx)
+	// import/export is backwards compatible, and we must write code specific to the verison we're dealing with.
+	if dep.Version != 3 {
```

Comment:
It feels like this check should happen in `stack.{Export,Import}` instead of putting the onus on the user

Reply:
Without generics, I think this is the best we get if we want this API to be backwards and forwards compatible. We could potentially add helper methods `ImportV1/ImportV2/ImportV3`, but I'd prefer to do the minimum here for now.